cluster/kubernetes/fakedynamicclient_test.go

```diff
@@ -0,0 +1,387 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubernetes
+
+/*
+This file was obtained from
+https://github.com/kubernetes/client-go/blob/4b43750b963d2b6e0f7527fe558e71c47bfc5045/dynamic/fake/simple.go
+and modified in the following way(s):
+
+ - the package was changed to `kubernetes`
+
+This file is here because it has a fix for
+https://github.com/kubernetes/client-go/issues/465, which is included
+in client-go v9, which we're not able to vendor at this time.
```

Comment:
What prevents us from vendoring it now (in a separate PR)? (in fact, we should probably bump it to version 10)

Reply:
Also, doesn't this belong to a separate package (e.g. `testutils`) we invoke form the tests? (I have been seeing a need for the same at #1668 )

Reply:
My thought process, no doubt: "ugh this is too much to explain right now, I'm sure I'll be able to figure out again what it means later"

Reply:
Most likely, depending on v9 or v10 either
 1. breaks compatibility with older API servers, or
 2. leads to an unresolvable clash with some other dependency